lib/releaselib.sh

```diff
@@ -803,52 +807,121 @@ release::docker::release () {
     "hyperkube"
   )
 
-  [[ "$registry" =~ gcr.io/ ]] && docker_push_cmd=("$GCLOUD" "docker" "--")
-
   # Activate credentials to push to gcr.io
   [[ "$registry" == "gcr.io/google_containers" ]] \
     && logrun $GCLOUD config set account $G_AUTH_USER
 
-  logecho
-  logecho "Send docker containers to $registry..."
-
-  # 'gcloud docker' gives lots of internal_failure's so add retries to
-  # all of the invocations
-  for arch in "${KUBE_SERVER_PLATFORMS[@]##*/}"; do
-    for binary in "${binaries[@]}"; do
-      docker_target="$binary-$arch:$version"
-      if ! logrun -r 5 ${docker_push_cmd[@]} \
-                       history "$registry/$docker_target"; then
-        logecho "$WARNING - Skipping non-existent $docker_target..."
-        continue
-      fi
-
-      logecho "Release $docker_target:"
-      logecho -n "- Pushing: "
-      logrun -r 5 -s ${docker_push_cmd[@]} push "$registry/$docker_target"
-
-      # If we have a amd64 docker image. Tag it without -amd64 also
-      # and push it for compatibility with earlier versions
-      if [[ $arch == "amd64" ]]; then
-        legacy_docker_target="$binary:$version"
-        logecho "Release legacy $legacy_docker_target:"
+  if [[ -d "$release_images" ]]; then
+    release::docker::release_from_tarfiles $* || ret=$?
+  else
+    # TODO: remove this when all kubernetes releases produce a release-images
+    # directory
+    [[ "$registry" =~ gcr.io/ ]] && docker_push_cmd=("$GCLOUD" "docker" "--")
 
-        logecho -n "- Tagging: "
-        logrun docker rmi "$registry/$legacy_docker_target" || true
-        logrun -r 5 -s docker tag "$registry/$docker_target" \
-                              "$registry/$legacy_docker_target" 2>/dev/null
+    logecho
```

Comment:
github does a terrible job displaying the diff, but I basically just indented here.